docs/extensions/tutorials/_posts/1970-01-07-ModifyingRootApp.md

```diff
@@ -0,0 +1,197 @@
+
+```JavaScript
+#file: shoutem.code-push/app/build/index.js
+const { injectCodePushAndroid, injectCodePushIos } = require('./injectCodePush');
+
+exports.preBuild = function preBuild() {
+  injectCodePushAndroid();
+  injectCodePushIos();
+}
+```
+
+#### inject{ExtensionName}{MobileOSName} convention
+
+- name your functions `inject{ExtensionName}{MobileOSName}`
+- start a new block by adding a comment declaring which file is going to be modified
+- assign that file's path to a new const
+- use inject() and/or replace() to apply modifications
+- create another block if more changes are needed
+
+The following example follows all of the above convention rules.
+
+```JavScript
+// app/settings.properties mods
+const gradlePropertiesPath = getGradlePropertiesPath({ cwd: projectPath });
+inject(
+  gradlePropertiesPath,
+  ANCHORS.ANDROID.GRADLE.PROPERTIES,
+  codepush.android.app.gradle.codepushKey,
+);
+
+// MainApplication.java mods
+const mainApplicationPath = getMainApplicationPath({ cwd: projectPath });
+inject(
+  mainApplicationPath,
+  ANCHORS.ANDROID.MAIN_APPLICATION.IMPORT,
+  codepush.android.app.import,
+);
+inject(
+  mainApplicationPath,
+  ANCHORS.ANDROID.MAIN_APPLICATION.RN_HOST_BODY,
+  codepush.android.app.rnHost,
+);
+inject(
+  mainApplicationPath,
+  ANCHORS.ANDROID.MAIN_APPLICATION.GET_PACKAGES,
+  codepush.android.app.getPackages,
+);
+```
+
+#### inject() and replace()
+
+The `inject()` and `replace()` functions can be used to either inject code at an anchor, or replace content in a specific file. Both functions will check if the code is already injected/replaced before doing so.
+
+**inject()** accepts the following arguments:
+
+- `filePath:` the path to the file that you need to modify
+- `anchor:` position in the file specified in the `ANCHORS` object
+- `contents`: the source code that needs to be injected at anchor position
+
+**replace()** accepts the following arguments:
+
+- `filePath:` the path to the file that you need to modify
+- `oldContent:` the source code to search for in the file
+- `newContent`: the source code that should replace `oldContent` in the file
+
+
+## Merging files
+
+As mentioned in the introduction, another tool provided for modifying the root app with an extension is to utilize the app segment merging.
+
+An example of this can be seen in our `shoutem.places` [extension](https://github.com/shoutem/extensions/tree/master/shoutem.places/app/android) in the `shoutem.places/app/android` directory. It merges, among other things, `AndroidManifest.xml` in order to add permissions for Internet and Location:
+
+```XML
+#file: shoutem.places/app/android/src/main/AndroidManifest.xml
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.shoutem.places">
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
+</manifest>
+```
```

Comment:
Please refer to my comment above for build.gradle, because merging is part of the build process for android modules. Explain that there is no similar mechanism for iOS, but we added a Merge Info.plist build step that merges all Info.plist files in extension/ios directory. The script is here: https://github.com/shoutem/platform/blob/develop/scripts/merge-info-plists.js and it's trying to mimic Android.